wrapper.js

```diff
@@ -31,13 +31,35 @@ function setupMethods (soljson) {
     };
   }
 
+  var alloc;
+  if ('_solidity_alloc' in soljson) {
+    alloc = soljson.cwrap('solidity_alloc', 'number', [ 'number' ]);
+  }
+
+  var reset;
+  if ('_solidity_reset' in soljson) {
+    reset = soljson.cwrap('solidity_reset', null, []);
+  }
+
   var copyToCString = function (str, ptr) {
     var length = soljson.lengthBytesUTF8(str);
     // This is allocating memory using solc's allocator.
-    // Assuming copyToCString is only used in the context of wrapCallback, solc will free these pointers.
-    // See https://github.com/ethereum/solidity/blob/v0.5.13/libsolc/libsolc.h#L37-L40
-    var buffer = soljson._malloc(length + 1);
-    soljson.stringToUTF8(str, buffer, length + 1);
+    //
+    // Before 0.6.0:
+    //   Assuming copyToCString is only used in the context of wrapCallback, solc will free these pointers.
+    //   See https://github.com/ethereum/solidity/blob/v0.5.13/libsolc/libsolc.h#L37-L40
+    //
+    // After 0.6.0:
+    //   The duty is on solc-js to free these pointers. We accomplish that by calling `reset` at the end.
+    var buffer;
+    if (isVersion6) {
+      buffer = alloc(length + 1);
```

Comment:
To complement my comment on the solidity PR: We can just use ``alloc(length)`` here, while still passing ``length + 1`` to ``stringToUTF8`` below and it's guaranteed to be valid, while then the ``std::string`` on the solidity side won't contain an additional zero.